### DISCUSSION TITLE
Use pyannote for realtime audio streaming with websocket.

### DISCUSSION THREAD

**Discussion by Aliw7979** [pyannote/pyannote-audio#1278] [Q&A] [answered]
Hello, I implemented a websocket server and it receives realtime audio chunks from user in bytes.
Is there any way to send bytes of audio to pyannote for speaker diarization?
Mention that I used diart for this purpose but it cant support multiple user.
Even [this link ](https://herve.niderb.fr/fastpages/2021/08/05/Streaming-voice-activity-detection-with-pyannote.html) from other similar discussions but couldn't help me either, because it only used voice activity detection, but I need to get something like this for result : 
 start=0.2s stop=1.5s speaker_0
 start=1.8s stop=3.9s speaker_1
 start=4.2s stop=5.7s speaker_0
Thanks.

**Comment by hbredin** [comment, marked as answer]
Nope -- pyannote.audio does not support streaming speaker diarization for now. 
`diart` is the way to go for now (not sure what you mean by "can't support multiple user").

**Reply by Aliw7979** [reply]
Thank you buddy, the problem with diart is something that I should ask from their rep & maybe it is irrelevant for here 
(In a nutshell, main problem that I faced with is their websocket source to receive audio chunks).
By the way, is there any plan to add streaming speaker to pyannote in 2023?
